examples/controls_example/public/react_controls/data_controls/initialize_data_control.tsx

```diff
@@ -0,0 +1,140 @@
+ */
+
+import { BehaviorSubject, combineLatestWith, distinctUntilChanged, mergeMap } from 'rxjs';
+
+import { CoreStart } from '@kbn/core-lifecycle-browser';
+import { DataView, DATA_VIEW_SAVED_OBJECT_TYPE } from '@kbn/data-views-plugin/common';
+import { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { Filter } from '@kbn/es-query';
+import { SerializedPanelState } from '@kbn/presentation-containers';
+import { StateComparators } from '@kbn/presentation-publishing';
+
+import { ControlGroupApi } from '../control_group/types';
+import { initializeDefaultControlApi } from '../initialize_default_control_api';
+import { ControlApiInitialization, ControlStateManager, DefaultControlState } from '../types';
+import { openDataControlEditor } from './open_data_control_editor';
+import { DataControlApi, DefaultDataControlState } from './types';
+
+export const initializeDataControl = <EditorState extends object = {}>(
+  controlId: string,
+  controlType: string,
+  state: DefaultDataControlState,
+  editorStateManager: ControlStateManager<EditorState>,
+  controlGroup: ControlGroupApi,
+  services: {
+    core: CoreStart;
+    dataViews: DataViewsPublicPluginStart;
+  }
+): {
+  dataControlApi: ControlApiInitialization<DataControlApi>;
+  dataControlComparators: StateComparators<DefaultDataControlState>;
+  dataControlStateManager: ControlStateManager<DefaultDataControlState>;
+  serializeDataControl: () => SerializedPanelState<DefaultControlState>;
+} => {
+  const {
+    defaultControlApi,
+    defaultControlComparators,
+    defaultControlStateManager,
+    serializeDefaultControl,
+  } = initializeDefaultControlApi(state);
+
+  const panelTitle = new BehaviorSubject<string | undefined>(state.title);
+  const defaultPanelTitle = new BehaviorSubject<string | undefined>(undefined);
+  const dataViewId = new BehaviorSubject<string>(state.dataViewId);
+  const fieldName = new BehaviorSubject<string>(state.fieldName);
+  const dataViews = new BehaviorSubject<DataView[] | undefined>(undefined);
+  const filters = new BehaviorSubject<Filter[] | undefined>(undefined);
+
+  const dataControlComparators: StateComparators<DefaultDataControlState> = {
+    ...defaultControlComparators,
+    title: [panelTitle, (value: string | undefined) => panelTitle.next(value)],
+    dataViewId: [dataViewId, (value: string) => dataViewId.next(value)],
+    fieldName: [fieldName, (value: string) => fieldName.next(value)],
+  };
+
+  const stateManager: ControlStateManager<DefaultDataControlState> = {
+    ...defaultControlStateManager,
+    dataViewId,
+    fieldName,
+    title: panelTitle,
+  };
+
+  /**
+   * Fetch the data view + field whenever the selected data view ID or field name changes; use the
+   * fetched field spec to set the default panel title, which is always equal to either the field
+   * name or the field's display name.
+   */
+  dataViewId
+    .pipe(
+      combineLatestWith(fieldName),
+      distinctUntilChanged(
```

Comment:
Do you need this `distinctUntilChanged` operator? Theoretically this will only get called when either the id or the field has changed.

Reply:
When I was using `mergeMap`, I was seeing a double emit when both data view and field name changed, and this equality check fixed that - but by switching to `switchMap`, I no longer see this. Fixed in e72127c7c6c407068636f427b5b5a5e3e3be42ab